src/components/Event.js

```diff
@@ -1,25 +1,70 @@
-import React, {Component} from 'react';
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+import { showPopup } from '../actions/popupActions';
+import { setEventProfile, setEventProfileById } from '../actions/eventActions';
+import store from '../store';
 
 class Event extends Component {
   constructor(props) {
-    super();
-    this.state = {
-      className: props.className,
-      place: props.place,
-      time: props.time,
-      name: props.name
-    };
+    super(props);
+    this.onShowPopup = this.onShowPopup.bind(this);
   }
 
+  componentDidMount() {
+    const url = window.location.pathname.split('/')[1].split('=');
+    const param = url[0];
+    const value = url[1];
+    if (param === 'eventId') {
+      const data = {
+        xPosCurrent: window.outerWidth / 2 - 120,
+        yPosCurrent: window.outerHeight / 2 - 50,
+        display: true,
+      };
+      this.props.onSetEventProfileById(value);
+      this.props.onShowPopup(data);
+    }
+  }
+
+  onShowPopup = function (event) {
```

Comment:
You can make this function "arrow function", then you don't need to bind it to "this"